src/rpdk/core/jsonutils/utils.py

```diff
@@ -185,6 +186,133 @@ def traverse_raw_schema(schema: dict, path: tuple):
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [{"baz": 1, "bin": 1}, {"baz": 2, "bin": 2}]}}, ("foo", "bar", "*", "baz"))
+    ([1, 2], [('foo', 'bar', 0, 'baz'), ('foo', 'bar', 1, 'baz')])
+    >>> traverse_path_for_sequence_members({}, ["foo"])
+    Traceback (most recent call last):
+    ...
+    KeyError: 'foo'
+    >>> traverse_path_for_sequence_members([], ["foo"])
+    Traceback (most recent call last):
+    ...
+    ValueError: invalid literal for int() with base 10: 'foo'
+    >>> traverse_path_for_sequence_members([], [0])
+    Traceback (most recent call last):
+    ...
+    IndexError: list index out of range
+    """
+    if path is None:
+        path = []
+    if not path_parts:
+        return [document], [tuple(path)]
+    path_parts = list(path_parts)
+    if not isinstance(document, Sequence):
+        return _handle_non_sequence_for_traverse(document, path_parts, path)
+    return _handle_sequence_for_traverse(document, path_parts, path)
+
+
+def _handle_non_sequence_for_traverse(
+    current_document: dict, current_path_parts: list, current_path: list
+) -> Tuple[List[object], List[tuple]]:
+    """
+    Handling a non-sequence member for `traverse_path_for_sequence_members` is like the loop block in `traverse`:
+
+    The next path part is the first part in the list of path parts.
+    The new document is obtained from the current document using the new path part as the key.
+    The next path part is added to the traversed path.
+
+    The traversal continues by recursively calling `traverse_path_for_sequence_members`
+    """
+    part_to_handle = current_path_parts.pop(0)
+    current_document = current_document[part_to_handle]
+    current_path.append(part_to_handle)
+    return traverse_path_for_sequence_members(
+        current_document, current_path_parts, current_path
+    )
+
+
+def _handle_sequence_for_traverse(
+    current_document: Sequence, current_path_parts: list, current_path: list
+) -> Tuple[List[object], List[tuple]]:
+    """
+    Check the new path part for the unpack sequence identifier (e.g. '*'), otherwise traverse index and continue:
+
+    The new document is obtained from the current document (a sequence) using the new path part as the index.
+    The next path part is added to the traversed path
+    """
+    sequence_part = current_path_parts.pop(0)
+    if sequence_part == UNPACK_SEQUENCE_IDENTIFIER:
+        return _handle_unpack_sequence_for_traverse(
+            current_document, current_path_parts, current_path
+        )
+    # otherwise, sequence part should be a valid index
+    current_sequence_part = int(sequence_part)
+    current_document = current_document[current_sequence_part]
+    current_path.append(current_sequence_part)
+    return [current_document], [tuple(current_path)]
+
+
+def _handle_unpack_sequence_for_traverse(
+    current_document, current_path_parts, current_path
```

Comment:
nit: `current_document: Sequence, current_path_parts: list, current_path: list`

Reply:
Whoops, missed the type annotations for this helper. I will add.